### ISSUE TITLE
Complemet `TestThreadedReceipts` flake?

### ISSUE THREAD

**Issue by DMRobertson** [matrix-org/synapse#15517]
https://github.com/matrix-org/synapse/actions/runs/4837501825/jobs/8621365637

**Comment by anoadragon453** [comment]
Indeed. https://github.com/matrix-org/synapse/actions/runs/4892228435/jobs/8733821785#step:7:5006

**Comment by DMRobertson** [comment]
https://github.com/matrix-org/synapse/actions/runs/4988864181/jobs/8932066016

**Comment by MadLittleMods** [comment]
https://github.com/matrix-org/synapse/actions/runs/5001049627/jobs/8959323881

**Comment by DMRobertson** [comment]
https://github.com/matrix-org/synapse/actions/runs/6695522291/job/18191386659
